src/snowcli/cli/snowpark/commands.py

```diff
@@ -0,0 +1,385 @@
+    remove_parameter_names,
+    check_if_replace_is_required,
+    build_udf_sproc_identifier,
+)
+from snowcli.cli.snowpark.manager import ProcedureManager, FunctionManager
+
+from snowcli.cli.snowpark_shared import (
+    CheckAnacondaForPyPiDependencies,
+    PackageNativeLibrariesOption,
+    PyPiDownloadOption,
+    snowpark_package,
+)
+from snowcli.cli.object.stage.manager import StageManager
+from snowcli.exception import ObjectAlreadyExistsError
+from snowcli.output.decorators import with_output
+from snowcli.output.types import (
+    MessageResult,
+    CommandResult,
+    SingleQueryResult,
+    QueryResult,
+    CollectionResult,
+)
+from snowcli.utils import (
+    get_snowflake_packages,
+)
+
+
+log = logging.getLogger(__name__)
+
+app = typer.Typer(
+    name="snowpark",
+    context_settings=DEFAULT_CONTEXT_SETTINGS,
+    help="Manage functions and procedures.",
+)
+
+
+LikeOption = typer.Option(
+    "%%",
+    "--like",
+    "-l",
+    help='Regular expression for filtering the procedure by name. For example, `list --like "my%"` lists all procedures in the **dev** (default) environment that begin with “my”.',
+)
+
+ReplaceOption = typer.Option(
+    False,
+    "--replace",
+    help="Replace procedure, even if no detected changes to metadata",
+)
+
+InstallCoverageWrapper = typer.Option(
+    False,
+    "--install-coverage-wrapper",
+    help="Whether to wrap the procedure with a code coverage measurement tool, so a coverage report can be later retrieved.",
+)
+
+ObjectTypeArgument = typer.Argument(
+    help="Type of snowpark object",
+    case_sensitive=False,
+)
+
+add_init_command(app, project_type="snowpark", template="default_snowpark")
+
+
+def _alter_procedure_artifact(
+    artifact_stage_path: str,
+    artifact_path: Path,
+    handler: str,
+    identifier: str,
+):
+    signature_start_index = identifier.index("(")
+    name = identifier[0:signature_start_index]
+    signature = identifier[signature_start_index:]
+    handler = _replace_handler_in_zip(
+        proc_name=name,
+        proc_signature=signature,
+        handler=handler,
+        coverage_reports_stage=DEPLOYMENT_STAGE,
+        coverage_reports_stage_path=f"/{artifact_stage_path}/coverage",
+        zip_file_path=str(artifact_path),
+    )
+
+    return handler
+
+
+@app.command("deploy")
+@with_output
+@global_options_with_connection
+def deploy(
+    install_coverage_wrapper: bool = InstallCoverageWrapper,
+    replace: bool = ReplaceOption,
+    **options,
+) -> CommandResult:
+    """Deploy procedures and functions defined in project."""
+    dm = DefinitionManager()
+    procedures = dm.project_definition.get("procedures", [])
+    functions = dm.project_definition.get("functions", [])
+
+    if not procedures and not functions:
+        raise ClickException(
+            "No procedures or functions were specified in project definition."
+        )
+
+    if (len(procedures) > 1 or functions) and install_coverage_wrapper:
```

Comment:
I think you should remove functions from `if`, now if I will have function and add flag coverage wrapper it will be true

Reply:
Yes, that's expected. We throw error if you have more than 1 procedure or any function and you provided coverage flags.

Reply:
But with 0 procedures, 1 function and flag you will throw error

Reply:
Yes, that's expected. We don't support coverage for functions

Reply:
Right :facepalm: